.github/workflows/rust.yml

```diff
@@ -12,8 +12,8 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  test:
-    name: Test all crates
+  test-relay:
```

Comment:
The intention of this job was that we can the same basic checks on all crates. For example:

- Running clippy on nightly in addition to stable will regularly give you CI errors as new lints are added.
- Formatting is sufficient to check in a single job.
- Checking for dead documentation links can be done once.

Reply:
There's a bunch of platform-specific code that we want to lint with clippy for connlib, I'd be open to completely discard compilations on nightly thou, since we won't be using any feature in nightly, it's a leftover from before.
I think for format to run the code-path must be available for that platform (although I need to double check), same for docs.

Reply:
Interesting, I hadn't considered that!

I've got a weak-spot for tuning CI pipelines so I might have a play at some point 😁